Tests/SentryTests/Integrations/Performance/IO/SentryFileIOTrackingIntegrationTests.swift

```diff
@@ -57,125 +69,285 @@ class SentryFileIOTrackingIntegrationTests: XCTestCase {
     }
     
     func test_WritingTrackingDisabled_forAutoPerformanceTrackingOption() {
+        // -- Act --
         SentrySDK.start(options: fixture.getOptions(enableAutoPerformanceTracing: false))
-        
+
+        // -- Assert --
         assertWriteWithNoSpans()
     }
     
     func test_WritingTrackingDisabled_TracingDisabled() {
+        // -- Act --
         SentrySDK.start(options: fixture.getOptions(tracesSampleRate: 0))
-        
+
+        // -- Assert --
         assertWriteWithNoSpans()
     }
     
-    func test_Writing_Tracking() throws {
-        if #available(iOS 18, macOS 15, tvOS 15, *) {
-            throw XCTSkip("File IO tracking for Swift.Data is disabled for this OS version")
+    func testData_Writing_Tracking() {
+        // -- Arrange --
+        let expectedSpanCount: Int
+        if #available(iOS 18, macOS 15, tvOS 18, *) {
+            // Automatic tracking of Swift.Data is not available starting with iOS 18, macOS 15, tvOS 15
+            // By asserting for it *not* working, we can lock down the expected behaviour and notice
+            // if it changes again in the future.
+            expectedSpanCount = 0
+        } else {
+            expectedSpanCount = 1
         }
+
+        // -- Act --
         SentrySDK.start(options: fixture.getOptions())
-        assertSpans(1, "file.write") {
+
+        // -- Assert --
+        assertSpans(expectedSpanCount, "file.write") {
             try? fixture.data.write(to: fixture.fileURL)
         }
     }
-    
-    func test_WritingWithOption_Tracking() throws {
-        if #available(iOS 18, macOS 15, tvOS 15, *) {
-            throw XCTSkip("File IO tracking for Swift.Data is disabled for this OS version")
+
+    func testData_WritingWithOption_Tracking() {
+        // -- Arrange --
+        let expectedSpanCount: Int
+        if #available(iOS 18, macOS 15, tvOS 18, *) {
+            // Automatic tracking of Swift.Data is not available starting with iOS 18, macOS 15, tvOS 15
+            // By asserting for it *not* working, we can lock down the expected behaviour and notice
+            // if it changes again in the future.
+            expectedSpanCount = 0
+        } else {
+            expectedSpanCount = 1
         }
+
+        // -- Act --
         SentrySDK.start(options: fixture.getOptions())
-        assertSpans(1, "file.write") {
+
+        // -- Assert --
+        assertSpans(expectedSpanCount, "file.write") {
             try? fixture.data.write(to: fixture.fileURL, options: .atomic)
         }
     }
+
+    func testDataExtension_Writing_Tracking() throws {
+        // -- Arrange --
+        // Automatic tracking of Swift.Data is not available starting with iOS 18, macOS 15, tvOS 15.
+        // Therefore, the extension method is only tested with these OS versions.
+        guard #available(iOS 18, macOS 15, tvOS 18, *) else {
+            throw XCTSkip("Data+SentryTracing is not tested on this OS version")
+        }
```

Comment:
`m`: I think we should still test it on these versions. Somebody might want to opt out of swizzling and use the manual approach. You can use it on older versions, so we should test it.

Reply:
I replaced the skip with different expected span counts instead. Will check if CI passes